main.js

```diff
@@ -126,8 +126,59 @@ function createNewActivity() {
+    if (pastActivities[i].minutes === 0 && pastActivities[i].seconds === 1) {
+      pastActivityCardHolder.innerHTML +=
+        `<div class="past-activity-card">
+          <div class="activity-color-indicator">
+            <div class="vertical-line vertical-line-${pastActivities[i].category.toLowerCase()}"></div>
+            <p class="past-activity-text">${pastActivities[i].category}</p>
+            <div class="min-sec-row">
+              <p class="min-sec-ptag"><span class="past-activity-minutes"></span><span class="past-activity-seconds"> ${pastActivities[i].seconds}</span> SECOND</p>
+            </div>
+          </div>
+          <p class="past-activity-description">${pastActivities[i].description}</p>
+        </div>`;
+    } else if (pastActivities[i].seconds === 1) {
+      pastActivityCardHolder.innerHTML +=
+        `<div class="past-activity-card">
+          <div class="activity-color-indicator">
+            <div class="vertical-line vertical-line-${pastActivities[i].category.toLowerCase()}"></div>
+            <p class="past-activity-text">${pastActivities[i].category}</p>
+            <div class="min-sec-row">
+              <p class="min-sec-ptag"><span class="past-activity-minutes">${pastActivities[i].minutes}</span> MIN<span class="past-activity-seconds"> ${pastActivities[i].seconds}</span> SECOND</p>
+            </div>
+          </div>
+          <p class="past-activity-description">${pastActivities[i].description}</p>
+        </div>`;
+    } else if (pastActivities[i].seconds === 0) {
+      pastActivityCardHolder.innerHTML +=
+        `<div class="past-activity-card">
+          <div class="activity-color-indicator">
+            <div class="vertical-line vertical-line-${pastActivities[i].category.toLowerCase()}"></div>
+            <p class="past-activity-text">${pastActivities[i].category}</p>
+            <div class="min-sec-row">
+              <p class="min-sec-ptag"><span class="past-activity-minutes">${pastActivities[i].minutes}</span> MIN <span class="past-activity-seconds"></span></p>
+            </div>
+          </div>
+          <p class="past-activity-description">${pastActivities[i].description}</p>
+        </div>`;
+    } else if (pastActivities[i].minutes === 0) {
+        pastActivityCardHolder.innerHTML +=
+          `<div class="past-activity-card">
+            <div class="activity-color-indicator">
+              <div class="vertical-line vertical-line-${pastActivities[i].category.toLowerCase()}"></div>
+              <p class="past-activity-text">${pastActivities[i].category}</p>
+              <div class="min-sec-row">
+                <p class="min-sec-ptag"><span class="past-activity-minutes"></span><span class="past-activity-seconds"> ${pastActivities[i].seconds}</span> SECONDS</p>
+              </div>
+            </div>
+            <p class="past-activity-description">${pastActivities[i].description}</p>
+          </div>`;
+    } else {
```

Comment:
I think the logic here is great! I'll test to make sure. You suggested that we might want to refactor this, and I think we can. If I have my thoughts straight, I think we can basically just write the html stuff once but then populate it with various things depending on the grammar situation. So maybe we'll have a separate function or functions like `determineMinutesWord` and `determineSecondsWord` or something like that, and have them return the right string for each case, which we can then interpolate into our html snippet at the right spot. I'll give it a try and keep y'all posted!